### ISSUE TITLE
bean 'taskScheduler' could not be registered. A bean with that name has already been defined

### ISSUE THREAD

**Issue by candrews** [lukas-krecan/ShedLock#122]
I'm using Spring Boot 2.1.2 and Shedlock 2.2.1.

Due to #121, I can't use `@EnableSchedulerLock` to configure shedlock. So I'm using the configuration approach documented for Shedlock 1.x: https://github.com/lukas-krecan/ShedLock/blob/shedlock-parent-1.2.0/README.md and it does not work.

Here's SchedulingConfig.java:
```
import java.time.Duration;

import javax.sql.DataSource;

import net.javacrumbs.shedlock.core.LockProvider;
import net.javacrumbs.shedlock.provider.jdbctemplate.JdbcTemplateLockProvider;
import net.javacrumbs.shedlock.spring.ScheduledLockConfiguration;
import net.javacrumbs.shedlock.spring.ScheduledLockConfigurationBuilder;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableScheduling;

/** Scheduling configuration and scheduled tasks.
 * @author Craig Andrews
 *
 */
@EnableScheduling
@Configuration
public class SchedulingConfig {
	@Value("${schedlock.schema}")
	private String schedLockSchema;

	@Bean
	public LockProvider lockProvider(final DataSource dataSource) {
		return new JdbcTemplateLockProvider(dataSource, schedLockSchema + ".shedlock");
	}

	@Bean
	public ScheduledLockConfiguration taskScheduler(final LockProvider lockProvider) {
		return ScheduledLockConfigurationBuilder
			.withLockProvider(lockProvider)
			.withPoolSize(10)
			.withDefaultLockAtMostFor(Duration.ofMinutes(10))
			.build();
	}

}
```

Here's the startup output leading up to the fatal error:
```
2019-01-30 17:14:41.880  INFO 3991 --- [  restartedMain] c.i.t.MyAppApplication       : Starting MyAppApplication on craigatwork with PID 3991 (/home/candrews/projects/myapp/target/classes started by candrews in /home/candrews/projects/myapp)
2019-01-30 17:14:41.883  INFO 3991 --- [  restartedMain] c.i.t.MyAppApplication       : No active profile set, falling back to default profiles: default
2019-01-30 17:14:42.128  INFO 3991 --- [  restartedMain] .e.DevToolsPropertyDefaultsPostProcessor : Devtools property defaults active! Set 'spring.devtools.add-properties' to 'false' to disable
2019-01-30 17:14:42.128  INFO 3991 --- [  restartedMain] .e.DevToolsPropertyDefaultsPostProcessor : For additional web related logging consider setting the 'logging.level.web' property to 'DEBUG'
2019-01-30 17:14:42.909  INFO 3991 --- [  restartedMain] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data repositories in DEFAULT mode.
2019-01-30 17:14:42.967  INFO 3991 --- [  restartedMain] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 53ms. Found 2 repository interfaces.
2019-01-30 17:14:43.002  WARN 3991 --- [  restartedMain] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.support.BeanDefinitionOverrideException: Invalid bean definition with name 'taskScheduler' defined in class path resource [org/springframework/boot/autoconfigure/task/TaskSchedulingAutoConfiguration.class]: Cannot register bean definition [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=org.springframework.boot.autoconfigure.task.TaskSchedulingAutoConfiguration; factoryMethodName=taskScheduler; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [org/springframework/boot/autoconfigure/task/TaskSchedulingAutoConfiguration.class]] for bean 'taskScheduler': There is already [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=schedulingConfig; factoryMethodName=taskScheduler; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [com/integralblue/myapp/SchedulingConfig.class]] bound.
2019-01-30 17:14:43.019  INFO 3991 --- [  restartedMain] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2019-01-30 17:14:43.021 ERROR 3991 --- [  restartedMain] o.s.b.d.LoggingFailureAnalysisReporter   : 

***************************
APPLICATION FAILED TO START
***************************

Description:

The bean 'taskScheduler', defined in class path resource [org/springframework/boot/autoconfigure/task/TaskSchedulingAutoConfiguration.class], could not be registered. A bean with that name has already been defined in class path resource [com/integralblue/myapp/SchedulingConfig.class] and overriding is disabled.

Action:

Consider renaming one of the beans or enabling overriding by setting spring.main.allow-bean-definition-overriding=true
```

I found a workaround is to modify ShedulingConfig to implement org.springframework.scheduling.annotation.SchedulingConfigurer (an empty implementation of the `configureTasks` method is fine). This prevents the autoconfiguration in `org.springframework.boot.autoconfigure.task.TaskSchedulingAutoConfiguration.taskScheduler(TaskSchedulerBuilder)` from running eliminating the other taskScheduler bean's creation.

**Comment by lukas-krecan** [comment]
Hi, thanks for feedback. This is exactly why  @EnableSchedulerLock has been introduced. It's hard to keep up with Spring Boot autoconfiguration magic and bugs using manual configuration. We should fix #121

**Comment by lukas-krecan** [comment]
The easiest way is to disable TaskSchedulingAutoConfiguration using this https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-auto-configuration.html#using-boot-disabling-specific-auto-configuration
